src/Maker/AbstractDockerMaker.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/*
+ * This file is part of the Symfony MakerBundle package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\MakerBundle\Maker;
+
+use Symfony\Bundle\MakerBundle\ConsoleStyle;
+use Symfony\Bundle\MakerBundle\DependencyBuilder;
+use Symfony\Bundle\MakerBundle\FileManager;
+use Symfony\Bundle\MakerBundle\Generator;
+use Symfony\Bundle\MakerBundle\Util\ComposeFileManipulator;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Yaml\Yaml;
+
+/**
+ * Convenient abstract class for docker makers.
+ *
+ * @author  Jesse Rushlow <jr@rushlow.dev>
+ *
+ * @internal
+ */
+abstract class AbstractDockerMaker extends AbstractMaker
+{
+    protected $fileManager;
+
+    /**
+     * @var ComposeFileManipulator
+     */
+    protected $composeFileManipulator;
+
+    /**
+     * @var string
+     */
+    protected $composeFilePath = '';
+
+    /**
+     * @var bool True if an existing docker-compose.yaml file was found.
+     */
+    protected $existingComposeFile = false;
+
+    /**
+     * @var string Service identifier to be set in docker-compose.yaml.
+     */
+    protected $serviceName = '';
+
+    /**
+     * @var string Version set in docker-compose.yaml for the service. e.g. latest
+     */
+    protected $serviceVersion = '';
```

Comment:
I think these might be better as protected methods. Then, if we need a property, we can default them to null. I know that might be tricky to get the flow right, but it smells weird how it is now.

Reply:
what did we decided to do with this? keep as is to "future proof" for typed props or refactor?